erizo_controller/erizoJS/models/Connection.js

```diff
@@ -52,27 +67,91 @@ class Connection {
       global.config.erizo.turnpass,
       global.config.erizo.networkinterface);
 
+    if (this.metadata) {
+        wrtc.setMetadata(JSON.stringify(this.metadata));
+    }
     return wrtc;
   }
 
-  _createMediaStream (id) {
+  _createMediaStream (id, mediaStreamOptions) {
     log.debug(`message: _createMediaStream, connectionId: ${this.id}, mediaStreamId: ${id}`);
     const mediaStream = new addon.MediaStream(this.threadPool, this.wrtc, id,
       this._getMediaConfiguration(this.mediaConfiguration));
     mediaStream.id = id;
+    if (mediaStreamOptions) {
+      mediaStream.metadata = mediaStreamOptions;
+      mediaStream.setMetadata(JSON.stringify(mediaStreamOptions));
+    }
     return mediaStream;
   }
 
-  addMediaStream(id) {
+  init() {
+    if (this.initialized) {
+      return;
+    }
+    this.initialized = true;
+    log.debug(`message: Init Connection, connectionId: ${this.id} `+
+              `${logger.objectToLog(this.options)}`);
+
+    this.wrtc.init((newStatus, mess) => {
+      log.info('message: WebRtcConnection status update, ' +
+               'id: ' + this.id + ', status: ' + newStatus +
+                ', ' + logger.objectToLog(this.metadata));
+      switch(newStatus) {
+        case CONN_INITIAL:
+          this.emit('status_event', {type: 'started'}, newStatus);
+          break;
+
+        case CONN_SDP:
+        case CONN_GATHERED:
+          this.wrtc.localDescription = new SessionDescription(this.wrtc.getLocalDescription());
+          const sdp = this.wrtc.localDescription.getSdp();
+          mess = sdp.toString();
+          mess = mess.replace(this.privateRegexp, this.publicIP);
```

Comment:
should probably be `this.options.privateRegexp` and `this.options.publicIP`

Reply:
good catch!